Feature/create home vc issue#3

suho opened the pull request:
- Finished Side Menu #4

suho committed:
#3: Create Home View Controller With 2 Button
#4: Create Empty Slide Menu
#5: Custom Navigation Bar

Add Library LGSideMenuController To Make Slide Menu

suho committed:
#4: Create 2 Custom Cell For Slide Bar

suho committed:
#4: Use Enum To Set Data For Table View In Slide Menu

suho committed:
#4: Finish Auto Layout Of Slide Menu

suho committed:
#4: Creat File Favorite To Make Action Of Side Menu

suho committed:
#4: Finished Send Menu Items From Side Menu To Background View Controller

suho committed:
Finished Side Menu - Issue #4

le-binh commented on FourSquare/Controllers/Background/BackgroundViewController.swift:
```diff
@@ -0,0 +1,65 @@
+
+    var leftViewController = LeftSideMenuViewController()
+
+    var allMenuItems = [ItemMenu]()
```
```
var menuItems: [ItemMenu] = []
```

suho replied:
fixed

le-binh commented on FourSquare/Controllers/AppDelegate/AppDelegate.swift:
```diff
@@ -18,19 +18,24 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         setup()
         window = UIWindow(frame: UIScreen.mainScreen().bounds)
-        window?.rootViewController = LoginViewController.vc()
+        let homeViewController = HomeViewController.vc()
```
make a function for creating rootViewController

suho replied:
fixed

le-binh commented on FourSquare/Controllers/Background/BackgroundViewController.swift:
```diff
@@ -0,0 +1,65 @@
+            let item = ItemMenu(item: MenuItemsSlide(rawValue: i)!, active: false)
+            allMenuItems.append(item)
+        }
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(self.UpdateItem), name: "ChangeItem", object: nil)
```
- `UpdateItem`: function name starts with lower case
- `ChangeItem`: please define it

suho replied:
fixed

le-binh commented on FourSquare/Controllers/Background/BackgroundViewController.swift:
```diff
@@ -0,0 +1,65 @@
+
+        super.rootViewController = initWithRootViewController
+        leftViewController = LeftSideMenuViewController.vc()
+        self.setLeftViewEnabledWithWidth(UIScreen.mainScreen().bounds.width * (2 / 3), presentationStyle: style, alwaysVisibleOptions: LGSideMenuAlwaysVisibleOptions.OnNone)
```
Make a define for `UIScreen.mainScreen().bounds.width * (2 / 3)`, for example: `leftSlideViewWidth`

suho replied:
fixed

le-binh commented on FourSquare/Controllers/Favorite/FavoriteViewController.swift:
```diff
@@ -0,0 +1,30 @@
+    @IBOutlet weak var navigationBarView: UIView!
+    @IBOutlet weak var sideMenuButton: UIButton!
+
+    override func viewDidLoad() {
```
delete function with empty implementation

suho replied:
fixed

le-binh commented on FourSquare/Controllers/LeftSideMenu/CustomCellSectionMainMenu/CustomMainMenuTableViewCell.swift:
```diff
@@ -0,0 +1,31 @@
+        // Configure the view for the selected state
+    }
+
+    func setUI(title: String, icon: UIImage) {
```
`setupUI`, `configureCell`

suho replied:
fixed

le-binh commented on FourSquare/Controllers/LeftSideMenu/CustomCellSectionMenuItems/CustomMenuItemsTableViewCell.swift:
```diff
@@ -0,0 +1,58 @@
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        self.setHeightForSwitch()
+        self.menuItemsActiveSwitch.addTarget(self, action: #selector(self.getActiveOfItem), forControlEvents: .ValueChanged)
```
drag IBAction

suho replied:
fixed

le-binh commented on FourSquare/Controllers/LeftSideMenu/CustomCellSectionMenuItems/CustomMenuItemsTableViewCell.swift:
```diff
@@ -0,0 +1,58 @@
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        self.setHeightForSwitch()
```
make `setupUI` and move all set-up functions to this function

suho replied:
fixed

le-binh commented on FourSquare/Controllers/LeftSideMenu/CustomCellSectionMenuItems/CustomMenuItemsTableViewCell.swift:
```diff
@@ -0,0 +1,58 @@
+
+}
+
+class ItemMenu: AnyObject {
```
`NSObject` instead of AnyObject

suho replied:
fixed

le-binh commented on FourSquare/Controllers/LeftSideMenu/CustomCellSectionMenuItems/CustomMenuItemsTableViewCell.swift:
```diff
@@ -0,0 +1,58 @@
+    func getActiveOfItem() {
+        if self.menuItemsActiveSwitch.on {
+            self.itemMenu.active = true
+            NSNotificationCenter.defaultCenter().postNotificationName("ChangeItem", object: nil, userInfo: ["item": self.itemMenu])
```
define `ChangeItem`, `item`

suho replied:
fixed

le-binh commented on FourSquare/Controllers/LeftSideMenu/CustomCellSectionMenuItems/CustomMenuItemsTableViewCell.swift:
```diff
@@ -0,0 +1,58 @@
+}
+
+class ItemMenu: AnyObject {
+    var item: MenuItemsSlide = MenuItemsSlide(rawValue: 0)!
```
remove default value

suho replied:
fixed

le-binh commented on FourSquare/Controllers/LeftSideMenu/LeftSideMenuViewController.swift:
```diff
@@ -0,0 +1,241 @@
+        case .MainMenu:
+            return ""
+        case .MenuItems:
+            return "Menu Items"
```
move string constant to Strings file

suho replied:
fixed

le-binh commented on FourSquare/Controllers/LeftSideMenu/LeftSideMenuViewController.swift:
```diff
@@ -0,0 +1,241 @@
+        }
+    }
+
+    var heightOfRow: CGFloat {
```
rowHeight

suho replied:
fixed

le-binh commented on FourSquare/Controllers/LeftSideMenu/LeftSideMenuViewController.swift:
```diff
@@ -0,0 +1,241 @@
+        }
+    }
+
+    var heightOfSection: CGFloat {
```
sectionHeight

suho replied:
fixed

suho committed:
Fixed Code After Review

suho committed:
Fix code
- Add Function getRootViewController in AppDelegate
- Change Define Color
- Change Class Item Menu

le-binh commented on FourSquare/Controllers/LeftSideMenu/LeftSideMenuViewController.swift:
```diff
@@ -0,0 +1,241 @@
+
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+
+        let slideMenuSection = SlideMenuSection(rawValue: indexPath.section)
```
use guard let here

suho replied:
fixed

le-binh commented on FourSquare/Controllers/LeftSideMenu/LeftSideMenuViewController.swift:
```diff
@@ -0,0 +1,241 @@
+
+            let mainMenuSlide = MainMenuSlide(rawValue: indexPath.row)
+
+            cell.configureCell(mainMenuSlide!.title, icon: mainMenuSlide!.icon)
```
Please do not use implicitly unwrapped optional

suho replied:
fixed

le-binh commented on FourSquare/Controllers/LeftSideMenu/LeftSideMenuViewController.swift:
```diff
@@ -0,0 +1,241 @@
+
+            let menuItemsSlide = MenuItemsSlide(rawValue: indexPath.row)
+
+            cell.configureCell(menuItemsSlide!)
```
Please do not use implicit unwrapped optional

suho replied:
fixed

le-binh commented on FourSquare/Define/SlideMenu.swift:
```diff
@@ -0,0 +1,15 @@
+import UIKit
+
+class SlideMenu {
+    static let getRootBackground = (UIApplication.sharedApplication().delegate?.window!!.rootViewController as? BackgroundViewController)!
```
Please do not use implicitly unwrapped optional
Do not make define for this. Make an extension
`getRootBackground`: please edit the name to make it clearer.

suho replied:
fixed

suho committed:
#3: Fix Code Review
- Use Guard For Ontional
- Make an Extension Of UIApplication

le-binh commented on FourSquare/Controllers/Background/BackgroundViewController.swift:
```diff
@@ -0,0 +1,65 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.leftViewSwipeGestureEnabled = false
+        for i in 0..<6 {
```
make a function for initializing menu items.
remember to break down code into functions

suho committed:
#3: Fixed Code Review

suho committed:
#3: Change Function Name
